### ISSUE TITLE
How to make "quiet" `fetch` request?

### ISSUE THREAD

**Issue by ryami333** [google/zx#405]
## Expected Behavior

When I wrap a `fetch` request in a `quiet`, I expect it not to log anything to the console. Eg:

```ts
const response = await quiet(fetch(url));
```

## Actual Behavior

The script logs the following to the console:

```zsh
$ fetch {url}
```

Is this behaviour deliberate, or is it a bug? Is there any way to make a "quiet" fetch request otherwise?

**Comment by antonmedv** [comment]
You can disable it with:
```js
$.verbose = false
```

I'm actually not happy with quiet() helper, and probably something better can be done after #397 merged.

**Comment by gustavopch** [comment]
The workaround posted by @antonmedv doesn't work well when you have concurrent fetches happening, where some need to be quiet and others don't. It would be nice to have something like `fetch.quiet()` or `fetch().quiet()`.

**Comment by antongolub** [comment]
Since v7 you can provide your own logger via `$.log` options, and filter out unnecessary events.
```js
export async function fetch(url: RequestInfo, init?: RequestInit) {
  $.log({ kind: 'fetch', url, init })
  return nodeFetch(url, init)
}
```

**Comment by gustavopch** [comment]
@antongolub Not sure if I follow how `$.log` would help. Here's a pseudo-code of what I was talking about:
```js
await Promise.all(
  // These will run concurrently, so a single global option isn't granular enough
  things.map(async thing => {
    await fetch(thing.someURL) // This one should be quiet
    await fetch(thing.otherURL) // This one should produce logs
  })
)
```

**Comment by antongolub** [comment]
```js
const stdLog = $.log

$.log = (ctx) => {
  const {kind, url} = log
  
  if (kind === 'fetch' && url === 'https://example.com' ) {
     return
  }
  
  return stdLog(ctx)
}

```

**Comment by gustavopch** [comment]
Thanks for the example, @antongolub. It's not as simple as `fetch.quiet()` or `fetch().quiet()` though. I mean, imagine if `$().quiet()` didn't exist and you had to pass a custom log function with logic to detect what logs to silence instead.

**Comment by antongolub** [comment]
```js
fetch.quiet = (...args) => within(() => {
  $.log = ...
  return fetch(...args)
})
```

**Comment by gustavopch** [comment]
Oh, that works well. Thanks, @antongolub.

**Comment by antonmedv** [comment]
No need to use builtin fetch. Just import directly. 

    import fetch from 'node-fetch'

    await fetch(url)
